config/initializers/z_plugin_parsers.rb

```diff
@@ -1,25 +1,19 @@
-class ParserRegistrator
-  def self.register_fact_parser(type, parser)
-    if (old_parser = FactParser.parser_for(type)) && old_parser != FactParser.parsers.default
-      Rails.logger.warn("WARNING: Parser #{old_parser} for type #{type} is replaced with #{parser}")
-    end
-
-    FactParser.register_fact_parser(type, parser)
-  end
-end
+# Puppet, the default parser
+Foreman::Plugin.fact_importer_registry.register(:puppet, PuppetFactImporter, true)
+Foreman::Plugin.fact_parser_registry.register(:puppet, PuppetFactParser, true)
 
 # Ansible
-Foreman::Plugin.fact_importer_registry.register(:ansible, ForemanAnsible::StructuredFactImporter, false)
-ParserRegistrator.register_fact_parser(:ansible, AnsibleFactParser)
+Foreman::Plugin.fact_importer_registry.register(:ansible, ForemanAnsible::StructuredFactImporter)
+Foreman::Plugin.fact_parser_registry.register(:ansible, AnsibleFactParser)
 
 # Katello
-::Foreman::Plugin.fact_importer_registry.register(Katello::RhsmFactName::FACT_TYPE, Katello::RhsmFactImporter)
-ParserRegistrator.register_fact_parser(Katello::RhsmFactName::FACT_TYPE, Katello::RhsmFactParser)
+Foreman::Plugin.fact_importer_registry.register(Katello::RhsmFactName::FACT_TYPE, Katello::RhsmFactImporter)
+Foreman::Plugin.fact_parser_registry.register(Katello::RhsmFactName::FACT_TYPE, Katello::RhsmFactParser)
 
 # Chef
 Foreman::Plugin.fact_importer_registry.register(:foreman_chef, ForemanChef::FactImporter)
-ParserRegistrator.register_fact_parser(:foreman_chef, ForemanChef::FactParser)
+Foreman::Plugin.fact_parser_registry.register(:foreman_chef, ForemanChef::FactParser)
 
 # Salt
 Foreman::Plugin.fact_importer_registry.register(:foreman_salt, ForemanSalt::FactImporter)
-FactParser.register_fact_parser(:foreman_salt, ForemanSalt::FactParser)
+Foreman::Plugin.fact_parser_registry.register(:foreman_salt, ForemanSalt::FactParser)
```

Comment:
We should probably move all parsers into a single module, this is scattered across the codebase. Feel free to do it in this PR, that would be appropriate change for sure.

Reply:
I'd prefer to redesign fact parsers so you don't need both an importer and a parser. That'd be also a good chance to move them. So I'd keep that for a separate PR. Right now it's perfectly scoped and easy to review.

Reply:
I'm fan of small changes (atomic ideally).I have the change module of parsers in mind too. I would like to do it in another PR. I was thinking to something about `Foreman::FactParsers::Puppet`...